pytest-embedded/pytest_embedded/dut.py

```diff
@@ -33,15 +33,70 @@ def write(self, *args, **kwargs) -> None:
         """
         self.pexpect_proc.write(*args, **kwargs)
 
-    def expect(self, *args, **kwargs) -> Match:
+    def expect(self, *args, **kwargs) -> Union[Match, AnyStr]:
         """
-        Expect from `pexpect_proc`. All arguments would pass to `pexpect.expect()`. Return the current `re.Match`
-        object if matched.
+        Expect from `pexpect_proc`. All the arguments would pass to `pexpect.expect()`.
+
+        Returns:
+            AnyStr: if you're matching pexpect.EOF or pexpect.TIMEOUT to get all the current buffers.
+
+        Returns:
+            re.Match: if matched given string.
         """
         try:
             self.pexpect_proc.expect(*args, **kwargs)
         except (pexpect.EOF, pexpect.TIMEOUT):
             logging.error(f'Not found {args}, {kwargs}')
             raise
         else:
+            if self.pexpect_proc.match in [pexpect.EOF, pexpect.TIMEOUT]:
+                return self.pexpect_proc.before.rstrip()
+
+            return self.pexpect_proc.match
+
+    def expect_exact(self, *args, **kwargs) -> Union[Match, AnyStr]:
+        """
+        Expect from `pexpect_proc`. All the arguments would pass to `pexpect.expect_exact()`.
+
+        Returns:
+            AnyStr: if you're matching pexpect.EOF or pexpect.TIMEOUT to get all the current buffers.
+
+        Returns:
+            re.Match: if matched given string.
+        """
+        try:
+            self.pexpect_proc.expect_exact(*args, **kwargs)
+        except (pexpect.EOF, pexpect.TIMEOUT):
+            logging.error(f'Not found {args}, {kwargs}')
+            raise
+        else:
+            if self.pexpect_proc.match in [pexpect.EOF, pexpect.TIMEOUT]:
+                return self.pexpect_proc.before.rstrip()
+
+            return self.pexpect_proc.match
+
+    def expect_list(self, *args, **kwargs) -> Union[Match, AnyStr]:
+        """
+        Expect from `pexpect_proc`. All the arguments would pass to `pexpect.expect_list()`.
+
+        Returns:
+            AnyStr: if you're matching pexpect.EOF or pexpect.TIMEOUT to get all the current buffers.
+
+        Returns:
+            re.Match: if matched given compiled regex.
+
+        Notes:
+            For the first argument, which is the regex list, you should pass the compiled regex with bytes.
```

Comment:
It looks a bit odd that this is the only case where we need to pass bytes rather than string as an argument?

Reply:
yeah... cause the [`expect_list`](https://pexpect.readthedocs.io/en/stable/api/pexpect.html#pexpect.spawn.expect_list) would take compiled regex as input, and we're using bytes in our buffer.

If we only pass compiled string regex, it would raise exception `TypeError: cannot use a string pattern on a bytes-like object`.